### PR TITLE
User can book 1 ticket at a time

### PR DESCRIPTION
**Pull request by Baonganbui123**

### CODE REVIEW

File: src/main/java/helpers/ElementHelpers.java [hunk 1]
```diff
@@ -9,9 +9,9 @@ public static void scrollToElement(WebElement element) {
         js.executeScript("arguments[0].scrollIntoView (true);", element);
     }
 
-    public static boolean isElementDisplayed(WebElement element){
+    public static boolean isElementDisplayed(By selector){
```
Comment (tanmle) [inline src/main/java/helpers/ElementHelpers.java:12]: why did you change this to selector instead of element?

File: src/test/java/tests/BookTicketTests.java [hunk 2]
```diff
@@ -1,27 +1,72 @@
...
+
+    @BeforeMethod(onlyForGroups = { "group 1" })
+    public void precondition_clickBookTicketTab() {
         LogHelpers.printInfo("Click on 'Book ticket' tab");
         bookTicketPage.clickBookTicketTab();
     }
 
-    @Test(description = "Login page displays when un-logged User clicks on 'Book ticket' tab")
+    @BeforeMethod(onlyForGroups = { "group 2" })
+    public void precondition_clickRegisterTab() {
```
Comment (tanmle) [inline src/test/java/tests/BookTicketTests.java:32]: your precondition is not just clicking on register tab

File: src/test/java/tests/BookTicketTests.java [hunk 3]
```diff
@@ -1,27 +1,72 @@
...
 
 public class BookTicketTests extends BaseTest {
 
     BookTicketPage bookTicketPage = new BookTicketPage();
+    RegisterPage registerPage = new RegisterPage();
     LoginPage loginPage = new LoginPage();
+    String newEmail = DataHelpers.getEmail();
     String headerLogin = "Login Page";
+    String successMsg = "Ticket Booked Successfully!";
+    String departDate = DataHelpers.getDateFromToday(6,"M/d/yyyy");
```
Comment (tanmle) [inline src/test/java/tests/BookTicketTests.java:22]: this is local var for just one test, you can specify inside the test itself

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/main/java/helpers/ElementHelpers.java (hunk 1 above)
Comment (Baonganbui123) [reply src/main/java/helpers/ElementHelpers.java:12]: Because in this case there is no element, it is not possible to get WebElement.
should use WebElement will not run, causing error

File: src/test/java/tests/BookTicketTests.java (hunk 2 above)
Comment (Baonganbui123) [reply src/test/java/tests/BookTicketTests.java]: I fixed, please check again!

File: src/test/java/tests/BookTicketTests.java (hunk 3 above)
Comment (Baonganbui123) [reply src/test/java/tests/BookTicketTests.java]: I fixed, please check again!

**Review by tanmle — APPROVED** [review]

**Review by MinhQuan119 — APPROVED** [review]
